include/seqan3/io/sequence_file/input.hpp

```diff
@@ -644,28 +642,100 @@ class sequence_file_input
+     * seqan3::sequence_file_input. It is not exposed to the user and allows to hide the implementation detail of
+     * storing a specfic format instance which is first known at runtime.
+     */
+    struct selected_input_format_base
+    {
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        selected_input_format_base() = default; //!< Default.
+        selected_input_format_base(selected_input_format_base const &) = default; //!< Default.
+        selected_input_format_base(selected_input_format_base &&) = default; //!< Default.
+        selected_input_format_base & operator=(selected_input_format_base const &) = default; //!< Default.
+        selected_input_format_base & operator=(selected_input_format_base &&) = default; //!< Default.
+        virtual ~selected_input_format_base() = default; //!< Virtual default.
+        //!\}
+
+        //!\brief The interface any given user input format must implement.
+        virtual void read_sequence_record(std::istream &, record_type &, sequence_file_input_options_type const &) = 0;
+    };
+
+    /*!\brief The specific selected format to read the records from.
+     *
+     * \tparam format_t The user specific format type to store.
+     *
+     * \details
+     *
+     * This class implements the format specific read operation based on the instantiated format type.
+     * The specfic format type is selected at runtime and stored through a pointer to the abstract
+     * seqan3::sequence_file_input::selected_input_format_base class. A virtual function call then ensures that the
+     * specific read record function of the selected format is invoked.
+     */
+    template <typename format_t>
+    struct selected_input_format final : public selected_input_format_base
+    {
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        selected_input_format() = default; //!< Default.
+        selected_input_format(selected_input_format const &) = default; //!< Default.
+        selected_input_format(selected_input_format &&) = default; //!< Default.
+        selected_input_format & operator=(selected_input_format const &) = default; //!< Default.
+        selected_input_format & operator=(selected_input_format &&) = default; //!< Default.
+        ~selected_input_format() = default; //!< Default.
+        //!\}
+
+        /*!\brief Reads the next format specific record from the given istream.
+         *
+         * \param[in, out] instream The input stream to extract the next record from.
+         * \param[in, out] record_buffer The record buffer to fill.
+         * \param[in] options User specific format options set from outside.
+         *
+         * \details
+         *
+         * Invokes the actual read sequence record function for the selected format and fills the record accordingly.
+         */
+        void read_sequence_record(std::istream & instream,
```

Comment:
Couldn't we add the documentation to the "interface" and just do here a `\override`?

Reply:
Maybe?